src/lib.rs

```diff
@@ -27,7 +27,13 @@ use tokio::net::UdpSocket;
 /// in a background task before making requests on the main thread.
 #[derive(Clone, Debug, Default)]
 pub struct Server {
-    store: HashMap<LowerName, Vec<IpAddr>>,
+    store: HashMap<LowerName, Response>,
+}
+
+#[derive(Clone, Debug)]
+pub enum Response {
+    Ok(Vec<IpAddr>),
+    Code(ResponseCode),
```

Comment:
This broadly looks like `Result<T, E>` to me, at least looking at the variants of `ResponseCode` in the [docs](https://docs.rs/hickory-proto/0.24.1/hickory_proto/op/response_code/enum.ResponseCode.html) they're all error types as far as I can see at a glance.

Does it make more sense to have:

```rust
type Response = Result<Vec<IpAddr>, ResponseCode>;
```

In terms of the API, I'd be tempted to rename `add_records` to just `add` and have it take something that can be converted into a response type. We'd need a simple trait (maybe `IntoResponse`) implemented for `Vec<IpAddr>` and `ResponseCode` but then users could call:

```rust
server.add("example.com", vec![...])?;
server.add("foo.example.com", ResponseCode::NXDomain)?;
```

It would be a breaking change, but given that it's just swapping `add_records` for `add` to allow it to be more generic maybe that's alright?